File: javascript/packages/highlighter/test/stylesheet.test.ts
import { describe, test, expect } from "vitest"

import { themes } from "../src/themes.js"
import { generateStylesheet } from "../src/stylesheet.js"

describe("Stylesheet", () => {
  test("generates custom properties for a single theme", () => {
    const css = generateStylesheet(themes.onedark, "onedark")

    expect(css).toMatchSnapshot()
  })

  test("generates a dark light pair", () => {
    const css = generateStylesheet(themes.onedark, "onedark:github-light", {
      scheme: themes["github-light"],
      label: "github-light",
    })

    expect(css).toMatchSnapshot()
  })

  test("maps named colors to hex", () => {
    const css = generateStylesheet(themes.simple, "simple")

    expect(css).toMatchSnapshot()
  })
})

File: rust/herb-highlighter/tests/stylesheet_test.rs
use herb_highlighter::stylesheet::generate_stylesheet;
use herb_highlighter::themes::{get_theme, Theme};

#[test]
fn generates_custom_properties_for_a_single_theme() {
  insta::assert_snapshot!(generate_stylesheet(get_theme(Theme::OneDark), "onedark", None));
}

#[test]
fn generates_a_dark_light_pair() {
  insta::assert_snapshot!(generate_stylesheet(
    get_theme(Theme::OneDark),
    "onedark:github-light",
    Some((get_theme(Theme::GithubLight), "github-light"))
  ));
}

#[test]
fn maps_named_colors_to_hex() {
  insta::assert_snapshot!(generate_stylesheet(get_theme(Theme::Simple), "simple", None));
}
